### PR TITLE
Add expand/collapse all actions into Task table

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Add Expand All/Collapse All context menu actions for Task table.

 ### Which part of the code base require a change
 
 - [x] Library: compiler
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [ ] Regression tests
 - [ ] Continous Integration (CI) scripts

### Screenshots
image

### CODE REVIEW

File: src/Compiler/TaskGlobal.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+static constexpr uint ExpandAreaRole = Qt::UserRole + 2;
+static constexpr uint RowVisibilityRole = Qt::UserRole + 3;
+static constexpr uint ParentDataRole = Qt::UserRole + 4;
+static constexpr uint TaskTypeRole = Qt::UserRole + 5;
+static constexpr uint TaskId = Qt::UserRole + 6;
+
+enum ExpandAreaAction {
+  Invert,
+  Expand,
+  Collaps,
```
Comment (skyler-rs) [inline src/Compiler/TaskGlobal.h:39]: Typo: Collaps -> Collapse

**Review by TarasSydor — APPROVED** [review]

File: src/Compiler/TaskGlobal.h (hunk 1 above)
Comment (KochynVolodymyr) [reply src/Compiler/TaskGlobal.h]: Fixed! Thanks

**Comment by ravic-rs** [comment]
@KochynVolodymyr changes looks good but there's a conflict to resolve.

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**